buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -45,13 +47,25 @@ class PrecommitTasks {
             forbiddenApisCliJar('de.thetaphi:forbiddenapis:2.6')
         }
 
+        Configuration jarHellConfig = project.configurations.create("jarHell")
+        if (project.findProject(":libs:elasticsearch-core") != null &&
```

Comment:
We should just use `ClasspathUtils.isElasticsearchProject()` here. We don't even really need to check if this is the core project either, It's perfectly fine for `JarHell` to depend on the core JAR, even in the core project itself as it'll just depend on the `jar` task so there's no risk in a cycle here.

Reply:
We do  need to exclude it, not for cycles, but for jar hell, adding the jar when the classes dir is also on the classpath via the source set when actually building the core project leads to jar hell for the JarHell class :)

Reply:
Right. The source set classpath uses the classes dir rather than the JAR. FWIW, if we used the `java-library` plugin the project dependency would _also_ use the classes dir instead of the JAR so there'd be no duplication but that's another story 😉